linking/cid/memorystorage.go

```diff
@@ -0,0 +1,51 @@
+package cidlink
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/ipld/go-ipld-prime/linking"
+)
+
+// Memory is a simple in-memory storage for cidlinks. It's the same as `storage.Memory`
+// but uses typical multihash semantics used when reading/writing cidlinks.
```

Comment:
```suggestion
// Memory is a simple in-memory storage for cidlinks. It's the same as `storage.Memory`
// but uses typical multihash semantics used when reading/writing cidlinks.
//
// Using multihash as the storage key rather than the whole CID will remove the
// distinction between CIDv0 and their CIDv1 counterpart. It also removes the
// distinction between CIDs where the multihash is the same but the codec is
// different, e.g. `dag-cbor` and a `raw` version of the same data.
```

I do wonder what the implications might be throughout the loading process in that latter case, where you request a `raw` block but it comes back as a `dag-cbor`. I know we're now doing this kind of thing IRL now (esp Estuary) and probably should make sure we're good with this kind of thing, but I wonder if there will be some surprises for us along the way.